src/main.rs

```diff
@@ -77,6 +78,21 @@ fn main() -> glib::ExitCode {
                 application_opts.test_dc = true;
             }
 
+            let client_id = dict.lookup::<i32>("client-id").unwrap();
+            let client_secret = dict.lookup::<String>("client-secret").unwrap();
```

Comment:
As said, we don't need turbofish syntax here ;-)

```suggestion
            let client_id = dict.lookup("client-id").unwrap();
            let client_secret = dict.lookup("client-secret").unwrap();
```

Reply:
appreciate your patience :+1: